app/src/lib/repository-matching.ts

```diff
@@ -92,3 +95,47 @@ export function matchExistingRepository(
     }) || null
   )
 }
+
+/**
+ * Check whether or not a GitHub repository matches a given remote.
+ *
+ * The remote's URL might match the cloneURL field of the repository (with a
+ * .git suffix) or it might match the htmlURL field (without a .git suffix)
+ *
+ * @param gitHubRepository the GitHub repository details found from the GitHub API
+ * @param remote the remote details found in the Git repository
+ */
+export function repositoryMatchesRemote(
+  gitHubRepository: GitHubRepository,
+  remote: IRemote
+): boolean {
+  const htmlUrl = gitHubRepository.htmlURL || ''
+
+  if (caseInsensitiveCompare(htmlUrl, remote.url) === 0) {
```

Comment:
While this happens to work here I don't think we should conflate comparison with equality. The comparison function is for sorting. Let's either do a manual lower case comparison here or add a dedicated `caseInsensitiveEquals` method in `equality.ts` and use that.

I'm assuming that we're doing this straight up equality check on htmlUrl here as a happy path? If so, shouldn't we be doing the same thing for cloneURL?